WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/StatsViewsWidget.kt

```diff
@@ -7,59 +7,122 @@ import android.content.Intent
 import android.net.Uri
 import android.os.Bundle
 import android.widget.RemoteViews
+import com.bumptech.glide.request.target.AppWidgetTarget
 import org.wordpress.android.R
 import org.wordpress.android.R.id
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsViewsWidgetConfigureViewModel.Color
+import org.wordpress.android.util.image.ImageManager
+import org.wordpress.android.util.image.ImageType.ICON
+import javax.inject.Inject
+
+const val SHOW_CHANGE_VALUE_KEY = "show_change_value_key"
+const val COLOR_MODE_KEY = "color_mode_key"
+const val SITE_ID_KEY = "site_id_key"
 
-/**
- * Implementation of App Widget functionality.
- * App Widget Configuration implemented in [StatsViewsWidgetConfigureActivity]
- */
 class StatsViewsWidget : AppWidgetProvider() {
+    @Inject lateinit var appPrefsWrapper: AppPrefsWrapper
+    @Inject lateinit var siteStore: SiteStore
+    @Inject lateinit var imageManager: ImageManager
+
+    override fun onReceive(context: Context, intent: Intent?) {
+        super.onReceive(context, intent)
+        (context.applicationContext as WordPress).component().inject(this)
+    }
+
     override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+        (context.applicationContext as WordPress).component().inject(this)
         for (appWidgetId in appWidgetIds) {
+            val minWidth = appWidgetManager.getAppWidgetOptions(appWidgetId)
+                    .getInt(AppWidgetManager.OPTION_APPWIDGET_MIN_WIDTH, 300)
             updateAppWidget(
                     context,
                     appWidgetManager,
-                    appWidgetId
+                    appWidgetId,
+                    appPrefsWrapper,
+                    siteStore,
+                    imageManager,
+                    minWidth > 250
```

Comment:
`250` is used in a couple of places - should we extract it to a resource or constant?